libr/main/ravc2.c

```diff
@@ -174,19 +167,23 @@ R_API int r_main_ravc2(int argc, const char **argv) {
 			r_list_free (uncommitted);
 		}
 	} else if (!strcmp(action, "reset")) {
-		save = r_vc_reset(rvc);
+		save = rvc->reset (rvc);
 	} else if (!strcmp(action, "log")) {
-		RList *commits = r_vc_log(rvc);
-		RListIter *iter;
-		const char *commit;
-		r_list_foreach(commits, iter, commit) {
-			printf ("%s\n****\n", commit);
+		if (!rvc->print_commits (rvc)) {
+			save = false;
+		}
+		goto ret;
+	} else if (!strcmp (action, "clone")) {
+		free (rp);
+		if (opt.argc < 3) {
+			eprintf ("Usage: %s [src] [dst]\n", argv[0]);
+			return -1;
 		}
-		r_list_free (commits);
-		return 0;
+		return !rvc->clone (rvc, argv[2 + opt.ind]);
+	} else {
+		eprintf ("Incorrect command\n");
 	}
-	eprintf ("Incorrect command\n");
 ret:
-	r_vc_close (rvc, save);
+	rvc->close (rvc, save);
```

Comment:
nope. call the api. which should internally call rvc->close if not null and such